### PR TITLE
[#2838] Add accessibility and label_accessibility tourisme/outdoor

### PR DESCRIPTION
**Pull request by LePetitTim**

### CODE REVIEW

**Review by camillemonchicourt — APPROVED** [review]

File: geotrek/tourism/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -7,7 +7,7 @@ msgid ""
 msgstr ""
 "Project-Id-Version: \n"
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2022-01-11 16:02+0000\n"
+"POT-Creation-Date: 2022-02-01 11:43+0000\n"
 "PO-Revision-Date: 2015-10-22 14:33+0200\n"
 "Last-Translator: jean.lenormand\n"
```
Comment (submarcos) [inline geotrek/tourism/locale/de/LC_MESSAGES/django.po:12]:
```suggestion
"Last-Translator: \n"
```

File: geotrek/tourism/models.py [hunk 2]
```diff
@@ -294,6 +309,10 @@ class TouristicContent(ZoningPropertiesMixin, AddPropertyMixin, PublishableMixin
     portal = models.ManyToManyField('common.TargetPortal',
                                     blank=True, related_name='touristiccontents',
                                     verbose_name=_("Portal"))
+    accessibility = models.TextField(verbose_name=_("Accessibility"), blank=True)
+    label_accessibility = models.ForeignKey(LabelAccessibility, verbose_name=_("Label accessibility"),
+                                            on_delete=models.CASCADE, related_name='touristiccontents', blank=True,
```
Comment (submarcos) [inline geotrek/tourism/models.py:314]:
```suggestion
                                            on_delete=models.PROTECT, related_name='touristiccontents', blank=True,
```

**Review by submarcos — COMMENTED** [review]
add select_related for endpoints / listviews using label_accessibility

Take care of CASCADE

File: geotrek/api/v2/serializers.py [hunk 3]
```diff
@@ -360,6 +370,7 @@ class Meta:
 
     class TouristicModelSerializer(PDFSerializerMixin, DynamicFieldsMixin, serializers.ModelSerializer):
         geometry = geo_serializers.GeometryField(read_only=True, source="geom_transformed", precision=7)
+        accessibility = serializers.SerializerMethodField(read_only=True)
```
Comment (submarcos) [inline geotrek/api/v2/serializers.py:373]:
```suggestion
        accessibility = serializers.SerializerMethodField()
```
stop to set read_only=True. SerializerMethodField has already this attribute to True. Then Viewsets are ReadOnly

File: geotrek/api/v2/serializers.py [hunk 4]
```diff
@@ -386,16 +400,16 @@ def get_description_teaser(self, obj):
             return get_translation_or_dict('description_teaser', self, obj)
 
     class TouristicContentSerializer(TouristicModelSerializer):
-        url = HyperlinkedIdentityField(view_name='apiv2:touristiccontent-detail')
-        types = serializers.SerializerMethodField(read_only=True)
         attachments = AttachmentSerializer(many=True, source='sorted_attachments')
         departure_city = serializers.SerializerMethodField(read_only=True)
+        types = serializers.SerializerMethodField(read_only=True)
```
Comment (submarcos) [inline geotrek/api/v2/serializers.py:405]:
```suggestion
        types = serializers.SerializerMethodField()
```

File: geotrek/api/v2/views/tourism.py [hunk 5]
```diff
@@ -11,6 +11,12 @@
 from geotrek.tourism import models as tourism_models
 
 
+class LabelAccessibilityViewSet(api_viewsets.GeotrekViewSet):
+    filter_backends = api_viewsets.GeotrekViewSet.filter_backends
```
Comment (submarcos) [inline geotrek/api/v2/views/tourism.py:15]:
```suggestion
```

File: geotrek/api/v2/views/tourism.py [hunk 6]
```diff
@@ -36,16 +42,27 @@ class TouristicContentViewSet(api_viewsets.GeotrekGeometricViewset):
...
         return tourism_models.TouristicContent.objects.existing()\
-            .select_related('category', 'reservation_system') \
+            .select_related('category', 'reservation_system', 'label_accessibility') \
             .prefetch_related('source', 'themes', 'type1', 'type2') \
             .annotate(geom_transformed=Transform(F('geom'), settings.API_SRID)) \
             .order_by('name')  # Required for reliable pagination
 
 
+class InformationDeskTypeViewSet(api_viewsets.GeotrekViewSet):
+    filter_backends = api_viewsets.GeotrekViewSet.filter_backends
```
Comment (submarcos) [inline geotrek/api/v2/views/tourism.py:52]:
```suggestion
```

File: geotrek/api/v2/views/tourism.py [hunk 7]
```diff
@@ -36,16 +42,27 @@ class TouristicContentViewSet(api_viewsets.GeotrekGeometricViewset):
...
 class InformationDeskViewSet(api_viewsets.GeotrekViewSet):
-    filter_backends = api_viewsets.GeotrekViewSet.filter_backends + (api_filters.TrekRelatedPortalFilter, api_filters.NearbyContentFilter,)
+    filter_backends = api_viewsets.GeotrekViewSet.filter_backends + (api_filters.TrekRelatedPortalFilter,
+                                                                     api_filters.NearbyContentFilter,
+                                                                     api_filters.GeotrekInformationDeskFilter)
     serializer_class = api_serializers.InformationDeskSerializer
-    queryset = tourism_models.InformationDesk.objects.order_by('name')
+
+    def get_queryset(self):
+        activate(self.request.GET.get('language'))
```
Comment (submarcos) [inline geotrek/api/v2/views/tourism.py:64]: is this really required ? why isn't it defined on others get_queryset ?

File: geotrek/api/v2/filters.py [hunk 8]
```diff
@@ -405,6 +405,35 @@ def _get_schema_fields(self, view):
...
 
 
+class GeotrekInformationDeskFilter(BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        qs = queryset
+        types = request.GET.get('types')
+        if types:
+            qs = qs.filter(type__in=types.split(','))
+        labels_accessibility = request.GET.get('labels_accessibility')
+        print(labels_accessibility)
```
Comment (submarcos) [inline geotrek/api/v2/filters.py:415]:
```suggestion
```

File: geotrek/api/v2/views/tourism.py (hunk 7 above)
Comment (LePetitTim) [reply geotrek/api/v2/views/tourism.py:64]: We need to translate for types and label_accessibility

**Review by submarcos — APPROVED** [review]

**Pull request merged**